### PR TITLE
Storing the results of executing "genesis scenarios" in the persistence

### PR DESCRIPTION
**Pull request by jakrawcz-rdx**
This is the first PR for the "expose Genesis Scenarios' results from Core API".
It only stores the results in the DB (and has no observable behavior yet - the API change is coming in another PR).

### CODE REVIEW

**Review by dhedey — APPROVED** [review]
Really nice, thanks!

File: core-rust/state-manager/src/state_manager.rs [hunk 1]
```diff
@@ -837,14 +790,93 @@ where
...
+    ) -> u32 {
+        let mut scenario = self
+            .find_scenario(epoch, nonce, scenario_name)
+            .unwrap_or_else(|| {
+                panic!(
+                    "Could not find scenario with logical name: {}",
+                    scenario_name
+                )
+            });
+        let scenario_base_version = genesis_commit_request_factory.state_version;
```
Comment (dhedey) [inline core-rust/state-manager/src/state_manager.rs:808]: I think this is a bit confusing that it is the state version _before_ the scenario executed.

I'd prefer to just set this to be:
```rust
    if let Some(scenario_start_state_version) = committed_transactions.first() {
        // We only commit a record of the scenario if it commits at least one transaction
        self.store.write().put(scenario_start_state_version, executed_scenario);
    }
```

File: core-rust/state-manager/src/store/rocks_db.rs [hunk 2]
```diff
@@ -133,12 +133,18 @@ enum RocksDBColumnFamily {
     /// Key: concat(account_address, state_version), value: null
     /// Given fast prefix iterator from RocksDB this emulates a Map<Account, Set<StateVersion>>
     AccountChangeStateVersions,
+    /// Additional details of "Scenarios" (and their transactions) executed as part of Genesis,
+    /// keyed by state version on top of which they are executed (note: this requires an assumption
```
Comment (dhedey) [inline core-rust/state-manager/src/store/rocks_db.rs:137]: I'm not sure we need this note, or maybe we can say that we explicitly store Scenarios which committed transactions.

**Review by dhedey — COMMENTED** [review]
Oh actually, just a quick thought

File: core-rust/state-manager/src/state_manager.rs (hunk 1 above)
Comment (jakrawcz-rdx) [reply core-rust/state-manager/src/state_manager.rs:808]: TBH a "state version after the first transaction that got committed as part of the scenario" is also a bit arbitrary to me :(
And in both cases, "not storing information on attempted scenarios which didn't successfully commit transactions" is also a bit imperfect.

Sooo... why do we want to identify these scenarios by a _transaction's_ ID anyway? Can we do "scenario order number" instead? Like `0 -> radiswap{...}; 1 -> transfer_xrd{...}`? That solves the above problems, and the contained transactions are easily identifiable anyways, and AFAIU we only list all these scenario executions (fully) from the endpoint (i.e. no-one queries them by any ID). Do you see any downsides of "scenario order number"?

Comment (dhedey) [reply core-rust/state-manager/src/state_manager.rs:808]: Sure - that sounds good :)

**Pull request merged**